docs/architecture/adr-008-evidence-module.md

```diff
@@ -0,0 +1,205 @@
+# ADR 008: Evidence Module
+
+## Changelog
+
+- 31-07-2019: Initial draft
+
+## Status
+
+Proposed
+
+## Context
+
+In order to support building highly secure, robust and interoperable blockchain
+applications, it is vital for the Cosmos SDK to expose a mechanism in which arbitrary
+evidence can be submitted, evaluated and verified resulting in some agreed upon
+penalty for any equivocations committed by a validator. Furthermore, such a
+mechanism is paramount for any [IBC](https://github.com/cosmos/ics/blob/master/ibc/1_IBC_ARCHITECTURE.md)
+protocol implementation in order to support the ability for any equivocations to
+be relayed back from a collateralized chain to a primary chain so that the
+equivocating validator(s) can be slashed.
+
+## Decision
+
+We will implement an evidence module in the Cosmos SDK supporting the following
+functionality:
+
+- Provide developers with the abstractions and interfaces necessary to define
+custom evidence messages and types along with their slashing penalties
+- Support the ability to route evidence message to handlers in any module to
+ determine the validity of a submitted equivocation
+- Support the ability through governance to modify slashing penalties of any
+evidence type
+- Querier implementation to support querying params, evidence types, params, and
+all submitted valid equivocations
+
+### Types
+
+First, we define the `Evidence` interface type. The `x/evidence` module may implement
+its own types that can be used by many chains (e.g. `CounterFactualEvidence`).
+In addition, other modules may implement their own `Evidence` types in a similar
+manner in which governance is extensible. It is important to note any concrete 
+type implementing the `Evidence` interface may include arbitrary fields such as
+an infraction time. We want the `Evidence` type to remain as flexible
+as possible.
+
+However, when submitting evidence to the `x/evidence` module, it must be submitted
+as an `Infraction` which includes mandatory fields outlined below.
+ 
+```go
+type Evidence interface {
+	Jailable() bool
+	Route() string
+	Type() string
+	ValidateBasic() sdk.Error
+	String() string
+}
+
+type Infraction struct {
+	Evidence
+	
+	ConsensusAddress    sdk.ConsAddress
+	InfractionHeight    int64
+	Power               int64
```

Comment:
is this the total validator power at the `InfractionHeight`? or is it the slashed power? a comment here would be helpful